src/calls/components/CallsView/Calendar.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import {
+    startOfMonth,
+    getDaysInMonth,
+    getDay,
+    getYear,
+    format,
+    addMonths,
+    subMonths,
+    addDays,
+    setMonth,
+    setYear
+} from 'date-fns';
+
+class Calendar extends Component {
+    state = {
+        date: new Date(),
+        months: [
+            'Jan',
+            'Feb',
+            'Mar',
+            'Apr',
+            'Maj',
+            'Jun',
+            'Jul',
+            'Avg',
+            'Sep',
+            'Okt',
+            'Nov',
+            'Dec'
+        ],
+        years: []
+    };
+
+    componentDidMount() {
+        this.props.handleChangeTableData(this.state.date);
+
+        if (this.state.years.length === 0) {
+            let years = [];
+            for (let y = 2000; y <= getYear(new Date()); y++) {
+                years.push(y);
+            }
+            this.setState({ years: years });
+        }
+    }
+
+    handleOnDateClick = event => {
+        let clickedDay = event.target.textContent;
+        let currentDate = this.state.date;
+        this.setState(
+            {
+                date: addDays(currentDate, clickedDay - currentDate.getDate())
+            },
+            () => this.props.handleChangeTableData(this.state.date)
+        );
+    };
+
+    handleChangeDate = event => {
+        let option = event.target.textContent;
+        if (option === '-') {
+            let currentDate = this.state.date;
+            this.setState({ date: subMonths(currentDate, 1) }, () =>
+                this.props.handleChangeTableData(this.state.date)
+            );
+        } else if (option === '+') {
+            let currentDate = this.state.date;
+            this.setState({ date: addMonths(currentDate, 1) }, () =>
+                this.props.handleChangeTableData(this.state.date)
+            );
+        } else {
+            if (!parseInt(option)) {
+                let currentDate = this.state.date;
+                this.setState(
+                    {
+                        date: setMonth(
+                            currentDate,
+                            this.state.months.indexOf(option)
+                        )
+                    },
+                    () => this.props.handleChangeTableData(this.state.date)
+                );
+            } else {
+                let currentDate = this.state.date;
```

Comment:
If you are going to grab `this.state.date` regardless of what the conditionals evaluate to, just grab it once at the beginning of the method body, and avoid repetition.